src/vw/src/main/scala/VowpalWabbitFeaturizer.scala

```diff
@@ -0,0 +1,140 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.ml.spark
+
+import com.microsoft.ml.spark.featurizer._
+import org.apache.spark.ml.Transformer
+import org.apache.spark.ml.param.{IntParam, ParamMap, StringArrayParam}
+import org.apache.spark.sql.types._
+import org.apache.spark.sql.{DataFrame, Dataset, Row}
+import org.apache.spark.sql.functions.{col, struct, udf}
+import org.vowpalwabbit.spark.VowpalWabbitMurmur
+import org.apache.spark.ml.linalg.Vectors
+import org.apache.spark.ml.util.{ComplexParamsReadable, ComplexParamsWritable, Identifiable}
+import org.apache.spark.ml.linalg.SQLDataTypes.VectorType
+
+import scala.collection.mutable.ArrayBuilder
+
+object VowpalWabbitFeaturizer extends ComplexParamsReadable[VowpalWabbitFeaturizer]
+
+class VowpalWabbitFeaturizer(override val uid: String) extends Transformer
+  with HasInputCols with HasOutputCol with HasNumBits with Wrappable with ComplexParamsWritable
+{
+  def this() = this(Identifiable.randomUID("VowpalWabbitFeaturizer"))
+
+  setDefault(inputCols -> Array())
+
+  val seed = new IntParam(this, "seed", "Hash seed")
+  setDefault(seed -> 0)
+
+  def getSeed: Int = $(seed)
+  def setSeed(value: Int): this.type = set(seed, value)
+
+  val stringSplitInputCols = new StringArrayParam(this, "stringSplitInputCols",
+    "Input cols that should be split add word boundaries")
+  setDefault(stringSplitInputCols -> Array())
+
+  def getStringSplitInputCols: Array[String] = $(stringSplitInputCols)
+  def setStringSplitInputCols(value: Array[String]): this.type = set(stringSplitInputCols, value)
+
+  private def getAllInputCols = getInputCols ++ getStringSplitInputCols
+
+  private def getFeaturizer(name: String, dataType: DataType, idx: Int, namespaceHash: Int): Featurizer = {
+    val stringSplitInputCols = getStringSplitInputCols
+
+    dataType match {
+      case DoubleType => new NumericFeaturizer(idx, name, namespaceHash, getMask, r => r.getDouble(idx))
```

Comment:
in the future, might be good to consider adding DateTime

Reply:
ah, looks like you have it in the TODO below already

Reply:
Any suggestions on how to featurize the date? For predictive scenarios one might wants to featurize into weekends vs weekday, am/pm, ...